app/src/main/java/com/bogdan/codeforceswatcher/features/auth/VerificationComposeActivity.kt

```diff
@@ -0,0 +1,202 @@
+package com.bogdan.codeforceswatcher.features.auth
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.MutableLiveData
+import com.bogdan.codeforceswatcher.CwApp
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.components.compose.*
+import com.bogdan.codeforceswatcher.components.compose.theme.AlgoismeTheme
+import io.xorum.codeforceswatcher.features.verification.redux.VerificationRequests
+import io.xorum.codeforceswatcher.features.verification.redux.VerificationState
+import io.xorum.codeforceswatcher.redux.store
+import tw.geothings.rekotlin.StoreSubscriber
+
+class VerificationComposeActivity : ComponentActivity(), StoreSubscriber<VerificationState> {
+
+    @ExperimentalComposeUiApi
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            AlgoismeTheme {
+                RestorePasswordScreen()
+            }
+        }
+    }
+
+    private val verificationState = MutableLiveData<VerificationState>()
+
+    @ExperimentalComposeUiApi
+    @Composable
+    private fun RestorePasswordScreen() {
+        val localFocusManager = LocalFocusManager.current
```

Comment:
Nitpick: I would group all `val`s at the beginning (without empty lines) for slightly better readability.